src/Features/CSharp/Portable/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider.cs

```diff
@@ -100,7 +100,14 @@ private static ExpressionSyntax ConvertInvocation(InvocationExpressionSyntax inv
 
             if (arguments.Count == 1)
             {
-                return arguments[0].Expression;
+                // Assignment expressions in a collection initializer will cause the compiler to 
+                // report an error.  This is because { a = b } is teh form for an object initializer,
+                // and the two forms are not allowed to mix/match.  Parenthesize the assignment to
+                // avoid the ambiguity.
+                var expression = arguments[0].Expression;
```

Comment:
What's the multi-argument scenario, and why doesn't it need something similar to this?